### PR TITLE
#319: Added PMD & SpotBugs plugins

### PR DESCRIPTION
**Pull request by oskarhinc**

### CODE REVIEW

File: src/main/java/org/benetech/mathshare/Application.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
+@SuppressWarnings("PMD.UseUtilityClass")
```
Comment (pawelcieszko) [inline src/main/java/org/benetech/mathshare/Application.java]: Is there any sensible possibility to bypass this warning?

Comment (oskarhinc) [reply src/main/java/org/benetech/mathshare/Application.java]: Unfortunately, the PMD, just like Checkstyle, treats our spring main class as a utility class - it has only public, static methods. Those classes should not expose public constructor, as they shouldn't be instantiated. In our case, we need this public constructor.

**Comment by pawelcieszko** [comment]
Resolve conflicts

**Comment by oskarhinc** [comment]
@pawelcieszko conflicts resolved

**Pull request merged**